client.go

```diff
@@ -448,22 +463,39 @@ func NewClient[TTx any](driver riverdriver.Driver[TTx], config *Config) (*Client
 			return nil, errMissingDatabasePoolWithQueues
 		}
 
-		// TODO: for now we only support a single instance per database/schema.
-		// If we want to provide isolation within a single database/schema,
-		// we'll need to add a config for this.
-		instanceName := "default"
-
 		client.completer = jobcompleter.NewBatchCompleter(archetype, driver.GetExecutor())
 		client.services = append(client.services, client.completer)
 
-		client.notifier = notifier.New(archetype, driver.GetListener(), client.monitor.SetNotifierStatus)
-		client.services = append(client.services, client.notifier)
+		// In poll only mode, we don't try to initialize a notifier that uses
+		// listen/notify. Instead, each service polls for changes it's
+		// interested in. e.g. Elector polls to see if leader has expired.
+		if !config.PollOnly {
+			client.notifier = notifier.New(archetype, driver.GetListener(), client.monitor.SetNotifierStatus)
+			client.services = append(client.services, client.notifier)
+		}
 
-		client.elector = leadership.NewElector(archetype, driver.GetExecutor(), client.notifier, instanceName, client.ID())
+		client.elector = leadership.NewElector(archetype, driver.GetExecutor(), client.notifier, &leadership.Config{
+			ClientID: config.ID,
+		})
 		client.services = append(client.services, client.elector)
 
-		if err := client.provisionProducers(); err != nil {
-			return nil, err
+		for queue, queueConfig := range config.Queues {
+			client.producersByQueueName[queue] = newProducer(archetype, driver.GetExecutor(), &producerConfig{
+				ClientID:          config.ID,
+				Completer:         client.completer,
+				ErrorHandler:      config.ErrorHandler,
+				FetchCooldown:     config.FetchCooldown,
+				FetchPollInterval: config.FetchPollInterval,
+				JobTimeout:        config.JobTimeout,
+				MaxWorkers:        queueConfig.MaxWorkers,
+				Notifier:          client.notifier,
+				Queue:             queue,
+				RetryPolicy:       config.RetryPolicy,
+				SchedulerInterval: config.schedulerInterval,
+				StatusFunc:        client.monitor.SetProducerStatus,
+				Workers:           config.Workers,
+			})
+			client.monitor.InitializeProducerStatus(queue)
```

Comment:
Moved initializing the producers up into `New` here. The helper function was so small, and this makes seeing their dependencies super clear as they're initialized alongside all the other services.